src/main.rs

```diff
@@ -116,23 +116,36 @@ async fn main() {
             sess.set_blocking(true);
             let msg = parse_recv_msg(&hdr_buf);
 
-            // Now read the actual data
+            // Now read the actual data...
             chan.read_exact(&mut buf[..msg.len as usize]).unwrap();
 
-            // And process it
+            // Our enum may not be up to date
+            #[allow(unreachable_patterns)]
```

Comment:
Why?

Reply:
somebody could have updated their server software faster, for example

Reply:
Which pattern is reported as unreachable?

Reply:
`Ok(m) => writeln!(stdout(), "unknown message type {m:?}").unwrap(),`, as all other enum entries are now handled

Reply:
Should probably mark the `enum` `#[non_exhaustive]` then?

Reply:
I think we should just drop the `Ok(m)` case.  If the enum gets new option, the code should be updated in sync. If the client gets unknown message, the `try_into()` will return `Err(foo)`. So the clause really becomes unused and can be dropped.

Reply:
Yes for that case, this makes sense. However, if you expect/want to add new enum variants without publishing a breaking crate release, `#[non_exhaustive]` is the only way to go.

Reply:
let's not go into that direction.

Reply:
ugh right, the try_into change altered this logic.. how about this:

```
diff --git a/src/lib.rs b/src/lib.rs
index 858601211ee6..22db72f49e0f 100644
--- a/src/lib.rs
+++ b/src/lib.rs
@@ -5,6 +5,7 @@ use std::mem::size_of;
 
 #[repr(u8)]
 #[derive(Debug, PartialEq)]
+#[non_exhaustive]
 pub enum Sk8brdMsgs {
     MsgSelectBoard = 1,
     MsgConsole,
diff --git a/src/main.rs b/src/main.rs
index 1882d7d74dbc..8bca3d5b645e 100644
--- a/src/main.rs
+++ b/src/main.rs
@@ -121,8 +121,6 @@ async fn main() {
             // Now read the actual data...
             chan.read_exact(&mut buf[..msg.len as usize]).unwrap();
 
-            // Our enum may not be up to date
-            #[allow(unreachable_patterns)]
             // ..and process it
             match msg.r#type.try_into() {
                 Ok(Sk8brdMsgs::MsgSelectBoard) => {
@@ -149,9 +147,7 @@ async fn main() {
                 Ok(Sk8brdMsgs::MsgFastbootContinue) => (),
 
                 Ok(m) => writeln!(stdout(), "unknown message type {m:?}").unwrap(),
-                Err(e) => {
-                    writeln!(stdout(), "got error '{e}' while processing msg: {msg:?}").unwrap()
-                }
+                Err(_) => {}
             };
             sess.set_blocking(false);
         }
```

the error case does nothing , as try_into() now can't really fail translating a u8 into something that has a repr(u8), IIUC?

Reply:
` ```diff ` please for readability :)

It can still fail per your `match` statement, if an unknown value is encountered? You're just allowing yourself the ability to return new `Ok(m)` variants without breaking consumer crates.

Reply:
My 2c would be for getting an error if one of the message kinds is unhandled. If we add another kind, we can bump the semver. But note, I'm not a pro at Rust libraries.

@konradybcio I'd suggest to leave the Err() message in place. It warns us (developers) that there was an unhandled message type.

Reply:
Yes with `TryFrom` and `#[non_exhaustive]` you get effectively two cases:

```rust
match foo {
    Ok(m) => todo!("sk8brd library added a new message flag {m:?} that we should handle"),
    Err(e) => todo!("Peer sent an invalid value (`{e}`) that is either not yet mapped in sk8brd or invalid"),
}
```